app/src/main/res/layout/activity_main.xml

```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".MainActivity">
+
```

Comment:
Could you please add a button that when tapped triggers the API call, and populate the TextView with the resulting JSON?

I want to be able to test if this is blocking the UI thread, and that's the easiest setup to test.

Reply:
Let's do one button for each API - so 3 total.